crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -302,6 +279,51 @@ impl<'this, TX: DbTx<'this>> DatabaseProvider<'this, TX> {
         Ok(account_transtions)
     }
 
+    /// Get the mappings of log address and log topic to block numbers where they occurred.
+    /// This function walks over the receipts and constructs corresponding mappings for log fields.
+    ///
+    /// # Returns
+    ///
+    /// * Mapping of log address to block numbers where they occurred.
+    /// * Mapping of log topic to block numbers where they occurred.
+    /// * Number of receipts walked.
+    pub fn get_log_addresses_and_topics(
+        &self,
+        range: RangeInclusive<BlockNumber>,
+    ) -> std::result::Result<(LogAddressIndices, LogTopicIndices, u64), TransactionError> {
```

Comment:
let's make this part of a `trait LogIndexProvider`, so that we can make this easier for @joshieDo's broader Provider -> Trait refactor/cleanup.